sips/sip-02x/sip-02x-non-sequential-multisig-transactions.md

```diff
@@ -0,0 +1,152 @@
+
+This document proposes having each signer sign the transaction directly:
+
+```
+signature_1(tx), signature_2(tx), ..., signature_n(tx)
+```
+
+This would address all of the concerns listed above, and would not increase transaction size or make it easier to forge a signature
+
+## Examples
+
+Imagine a DAO that has a management team comprised of five members.
+They create a 3 out of 5 multisig account on Stacks.
+The existing multisig standard mandates that all transactions from this account be signed in an order which is established upon account creation.
+The ordering requirement creates a hierarchy where keys near the start of the sequence have more flexibility than those near the end. 
+To illustrate some of the limitations this creates:
+
+- In a scenario where the 1st member initiates a transaction and the 4th signs it, it prohibits the 2nd and 3rd members from signing. The responsibility then falls solely on the 5th member to finalize the transaction.
+- Once the 5th member has signed a transaction, no further signatures are possible.
+- If the 3rd member initiates a transaction, only the 4th and 5th members are eligible to provide subsequent signatures.
+- Initiating a transaction by the 4th or 5th member is impossible, as there are insufficient subsequent members to complete the signing process.
+
+To summarize, while such a multisig setup might suffice for smaller teams, as the number of required signers increases, it becomes increasingly difficult to create a transaction. This SIP aims to remove these limitations.
+
+# Specification
+
+This section should be interpreted as a patch to the existing "[Specification](https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#specification)" section of SIP-005.
+For anything not mentioned here, the rules from SIP-005 still apply.
+
+## Transactions
+
+### Transaction Encoding
+
+#### Transaction Authorization
+
+Add new hash modes `0x05` and `0x07`. These numbers were chosen in order to used in order to maintain the following relationships:
+ - `is_multisig = hash_mode & 0x1`
+ - `is_p2wsh_p2sh = hash_mode & 0x2`
+ - `is_non_sequential_multisig = hash_mode & 0x4`
+
+| Hash mode | Spending Condition | Mainnet version | Hash algorithm |
+| --------- | ------------------ | --------------- | -------------- |
+| `0x00` | Single-signature | 22 | Bitcoin P2PKH |
+| `0x01` | Multi-signature | 20 | Bitcoin redeem script P2SH |
+| `0x02` | Single-signature | 20 | Bitcoin P2WPK-P2SH |
+| `0x03` | Multi-signature | 20 | Bitcoin P2WSH-P2SH |
+| `0x05` | Non-sequential multi-signature | 20 | Bitcoin redeem script P2SH |
+| `0x07` | Non-sequential multi-signature | 20 | Bitcoin P2WSH-P2SH |
+
+#### Transaction Signing and Verifying
+
+The steps for signing a non-sequential multisig transaction (hash modes `0x05` and `0x07`) shall be as follows:
+
+0. Set the spending condition address, and optionally, its signature count.
```

Comment:
I think for SIPS/documents in general, we should start numbered lists with `1` rather than the *technically correct* `0`. 
Cosmetic change - I don't feel strongly if it remains starting at `0` though.

Reply:
I'd prefer to keep it at `0` *only* because this is a modification of SIP-05 and that uses 0 indexing for these steps

Reply:
I don't think the signature count should be optional.  Either this algorithm needs to work without it, or it should be mandatory.

Reply:
Agreed, but I don't the code actually treats it as optional. This line was copied directly from SIP-005, so we may want to revise that also